pkg/reportingplugins/mercury/aggregate_functions.go

```diff
@@ -56,11 +56,11 @@ func GetConsensusBid(paos []ParsedAttributedObservation, f int) (*big.Int, error
 }
 
 // GetConsensusAsk gets the median ask
-func GetConsensusAsk(paos []ParsedAttributedObservation, f int) (*big.Int, error) {
+func GetConsensusAsk(paos []ParsedObservation, f int) (*big.Int, error) {
 	var validAsks []*big.Int
 	for _, pao := range paos {
-		if pao.PricesValid {
-			validAsks = append(validAsks, pao.Ask)
+		if pao.GetPricesValid() {
```

Comment:
Since these are methods now there may be a cleaner way to represent this, e.g.

```
ask, valid := pao.GetAsk()
if valid {
  validAsks = append(validAsks, ask)
}
```